src/app/store/authors/authors.reducer.ts

```diff
@@ -0,0 +1,60 @@
+
+import { createReducer, on } from "@ngrx/store";
+import { Author } from "src/app/services/author";
+import { requestAuthors, 
+         requestAuthorsSuccess, 
+         requestAuthorsFail, 
+         requestAddAuthor, 
+         requestAddAuthorSuccess, 
+         requestAddAuthorFail,
+         resetAddedAuthor} from "./authors.actions";
+
+export const authorsFeatureKey = 'authorsFeatureKey';
+
+export interface AuthorsState {
+    authors: Author[],
+    addedAuthor: Author,
+    error?: string,
+};
+
+export const initialState: AuthorsState = {
+    authors: [],
+    addedAuthor: {name: ''},
+};
+
+export const authorsReducer = createReducer(
+    initialState,
+    on(requestAuthors, (state) => ({
+        ...state,
+    })),
+
+    on(requestAuthorsSuccess, (state, authors) => ({
+        ...state,
+        authors: authors.authors,
+    })),
+
+    on(requestAuthorsFail, (state, err) => ({
+        ...state,
+        error: err.error
+    })),
+
+    on(requestAddAuthor, (state, author) => ({
+        ...state,
+        author: author.author
+    })),
```

Comment:
```suggestion
    on(requestAddAuthor, (state, {author}) => ({
        ...state,
        author
    })),
``` 
Destructuring assignment may make our code more readable